google/cloud/storage/client_test.cc

```diff
@@ -221,6 +221,38 @@ TEST_F(ClientTest, LoggingDecoratorsRestClient) {
                           "RetryClient"));
 }
 
+/// @test Verify the constructor creates the right set of RawClient decorations.
+TEST_F(ClientTest, FullStack) {
+  // Reset environment variables. We want the `Options` parameter to control the
+  // structure of the stack.
+  ScopedEnvironment logging("CLOUD_STORAGE_ENABLE_TRACING", absl::nullopt);
+  ScopedEnvironment legacy("GOOGLE_CLOUD_CPP_STORAGE_USE_LEGACY_HTTP",
+                           absl::nullopt);
+
+  // Create a client. Use the anonymous credentials because on the CI
+  // environment there may not be other credentials configured.
+  auto options = Options{}
+                     .set<UnifiedCredentialsOption>(MakeInsecureCredentials())
+                     .set<internal::UseRestClientOption>(true)
+                     .set<TracingComponentsOption>({"raw-client"});
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+  options.set<experimental::OpenTelemetryTracingOption>(true);
```

Comment:
nit: our tests would be more self-consistent with:

```cc
#include "google/cloud/testing_util/opentelemetry_matchers.h"

using ::google::cloud::testing_util::EnableTracing;
options = EnableTracing(std::move(options);
```

Reply:
Okay.  But this option setting is odd...